src/sentry/web/frontend/organization_auth_settings.py

```diff
@@ -152,7 +167,23 @@ def handle_existing_provider(
             if form.initial != form.cleaned_data:
                 changed_data = {}
                 for key, value in form.cleaned_data.items():
-                    if form.initial.get(key) != value:
+                    if key == "x509cert":
+                        original_idp = form.initial.get("idp", {})
+                        if original_idp.get("x509cert", "") != value:
+                            auth_provider.config = {
+                                **auth_provider.config,
+                                "idp": {
+                                    **original_idp,
+                                    "x509cert": value,
+                                },
+                            }
+                            auth_service.update_provider_config(
+                                organization_id=organization.id,
+                                auth_provider_id=auth_provider.id,
+                                config=auth_provider.config,
+                            )
+                            changed_data["x509cert"] = f"to {value}"
```

Comment:
should this just be above and not in this loop? i see that's how the other values that get change end up getting used. (e.g. we just always call change_scim / update_default_role)

Reply:
Yea, although we'd still want to catch the audit log change, so if we're iterating through here anyways 🤷